extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -1477,18 +1477,31 @@ static public function onWatchArticle( &$user, &$article ) {
 		return true;
 	}
 
+	/**
+	 * @param User $user
+	 * @param Article $article
+	 *
+	 * @return bool
+	 */
 	static public function onUnwatchArticle( &$user, &$article ) {
-		$app = F::app();
 		$title = $article->getTitle();
 
-		if ( !empty( $app->wg->EnableWallExt ) && static::isWallMainPage( $title ) ) {
-			static::processActionOnWatchlist( $user, $title->getText(), 'remove' );
+		// Wall code makes up fake Title objects to trick MW into handling its "Thread:1234" style
+		// title text.  It also reuses the same namespaces for Forums, user walls, replies, etc.
+		// Since we need a real Title backed by a DB row, we need to reconstruct a title object
+		// if the current one looks fake.  If this already is a real Title the normal unwatch handling
+		// that called this hook will take care of this for us.
+		if ( ( $article->getId()  == 0 ) && preg_match( '/^(\d+)$/', $title->getText(), $matches ) ) {
```

Comment:
You could also use ```ctype_digit``` here to check the title is all digits, if you didn't want the overhead of regex. But I doubt in this case there's any performance hit.